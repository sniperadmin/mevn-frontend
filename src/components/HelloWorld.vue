<template>
  <v-container>
    <v-row class="text-center">
      <v-col cols="12">
        <Logos />
      </v-col>

      <v-col class="mb-2">
        <Card no-title>
          <v-card-text class="headline font-weight-bold">
            <div>MEVN-Auth APP</div>
            <v-avatar
              size="avatarSize"
              class="mt-9"
            >
              <img src="https://forums.coretabs.net/user_avatar/forums.coretabs.net/sniperadmin/120/2694_2.png" alt="nasr">
            </v-avatar>
            <h4 class="mt-7">
              This is a simple authentication/authorization app
              built with Mongo-Express-Vue-Node and boosted by
              <strong class="green--text">Unit testing with Jest</strong>
            </h4>
          </v-card-text>
        </Card>
      </v-col>
      <v-col>
        <v-alert
          outlined
          color="purple"
        >
          <div class="title">Instructions</div>
          <div>
            You can go to register or login to test authentication
          </div>
        </v-alert>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
import Card from './Card.vue'
  export default {
    name: 'HelloWorld',
    components: {
      Logos: () => import('@/components/Logos.vue'), Card
    },
    data: () => ({}),
  }
</script>
