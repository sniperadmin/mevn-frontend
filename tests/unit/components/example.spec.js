import Vue from 'vue'
import Vuetify from 'vuetify'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

describe('HelloWorld.vue', () => {
  let wrapper
  const localVue = createLocalVue()

  Vue.use(Vuetify)

  const vuetify = new Vuetify()

  beforeEach(() => {
    wrapper = shallowMount(HelloWorld, {
      localVue,
      vuetify
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('renders component with contents', () => {
    const h4 = wrapper.find('h4')
    const cardText = wrapper.find('v-card-text-stub').find('div')
    const alert = wrapper.find('v-alert-stub')

    expect(wrapper.html()).toMatchSnapshot()
    expect(h4.exists()).toBe(true)
    expect(h4.text()).toContain('authentication/authorization')
    expect(cardText.text()).toBe('MEVN-Auth APP')
    expect(alert.exists()).toBe(true)
    expect(alert.find('.title').text()).toBe('Instructions')
    expect(alert.text()).toContain('You can go to register or login to test authentication')
  })
})
