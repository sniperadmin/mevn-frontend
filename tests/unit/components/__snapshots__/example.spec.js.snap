// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders component with contents 1`] = `
<v-container-stub tag="div">
  <v-row-stub tag="div" class="text-center">
    <v-col-stub cols="12" tag="div">
      <v-row-stub tag="div" justify="center">
        <v-col-stub cols="2" sm="4" tag="div">
          <v-img-stub height="200" alt="Mongo logo" contain="true" options="[object Object]" position="center center" src="../assets/mongodb.png" transition="fade-transition" class="shrink mt-3"></v-img-stub>
        </v-col-stub>
        <v-col-stub cols="2" sm="4" tag="div">
          <v-img-stub height="200" alt="Express logo" contain="true" options="[object Object]" position="center center" src="../assets/express-min.png" transition="fade-transition" class="shrink mt-3"></v-img-stub>
        </v-col-stub>
        <v-col-stub cols="2" sm="4" tag="div">
          <v-img-stub height="200" contain="true" options="[object Object]" position="center center" src="../assets/logo.svg" transition="fade-transition"></v-img-stub>
        </v-col-stub>
        <v-col-stub cols="2" sm="4" tag="div">
          <v-img-stub height="200" alt="Vue logo" contain="true" options="[object Object]" position="center center" src="../assets/logo.png" transition="fade-transition" class="shrink mt-3"></v-img-stub>
        </v-col-stub>
        <v-col-stub cols="2" sm="4" tag="div">
          <v-img-stub height="200" alt="Nodejs logo" contain="true" options="[object Object]" position="center center" src="../assets/nodejs.png" transition="fade-transition" class="shrink mt-3"></v-img-stub>
        </v-col-stub>
        <v-col-stub cols="8" sm="4" tag="div">
          <v-img-stub height="200" alt="Nodejs logo" contain="true" options="[object Object]" position="center center" src="../assets/jest.png" transition="fade-transition" class="shrink mt-3"></v-img-stub>
        </v-col-stub>
      </v-row-stub>
    </v-col-stub>
    <v-col-stub tag="div" class="mb-2">
      <card-stub notitle="true">
        <v-card-text-stub class="headline font-weight-bold">
          <div>MEVN-Auth APP</div>
          <v-avatar-stub size="avatarSize" class="mt-9"><img src="https://forums.coretabs.net/user_avatar/forums.coretabs.net/sniperadmin/120/2694_2.png" alt="nasr"></v-avatar-stub>
          <h4 class="mt-7">
            This is a simple authentication/authorization app
            built with Mongo-Express-Vue-Node and boosted by
            <strong class="green--text">Unit testing with Jest</strong></h4>
        </v-card-text-stub>
      </card-stub>
    </v-col-stub>
    <v-col-stub tag="div">
      <v-alert-stub color="purple" tag="div" value="true" closelabel="$vuetify.close" icon="" outlined="true">
        <div class="title">Instructions</div>
        <div>
          You can go to register or login to test authentication
        </div>
      </v-alert-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
